openwisp_monitoring/device/base/models.py

```diff
@@ -454,6 +455,22 @@ def handle_disabled_organization(cls, organization_id):
             status='unknown'
         )
 
+    @classmethod
+    def handle_critical_metric(cls, instance, **kwargs):
+        from openwisp_monitoring.device.settings import get_critical_device_metrics
+
+        critical_metrics = [metric['key'] for metric in get_critical_device_metrics()]
+        if instance.check_type in critical_metrics:
+            try:
+                device_monitoring = cls.objects.get(device=instance.content_object)
+                if not instance.is_active or kwargs.get('signal') == post_delete:
+                    device_monitoring.update_status('unknown')
+                    health_status_changed.send(
```

Comment:
Isn't this already performed by the `update_status`? It seems redundant to me and hence I think it should be removed.